src/Approximations/overapproximate.jl

```diff
@@ -516,3 +516,69 @@ function overapproximate(cap::Intersection{N,
     return overapproximate(swap(cap), dir; kwargs...)
 end
 
+"""
+    overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa=Hyperrectangle)  where {N}
+
+Decompose a lazy linear map of a cartesian product array.
+
+### Input
+
+- `lm`   -- lazy linear map of cartesian product array
+- `CartesianProductArray` -- type for dispatch
+- `oa` -- approximation option for decomposition
+
+### Output
+
+A `CartesianProductArray` representing the decomposed linear map.
+"""
+function overapproximate(lm::LinearMap{N, <:CartesianProductArray{N,
+                            <:LazySet{N}}}, ::Type{<:CartesianProductArray}, oa=Hyperrectangle) where {N}
+
+    @assert size(lm.M, 2) == dim(lm.X) "matrix needs to be commensurate with the cartesian product"
+
+    cpa = lm.X
+    M = lm.M
+
+    col_end_ind = 0
+
+    array = allocate_result(oa, N)
+    sizehint!(array,length(cpa.array))
+
+    i, n = 1, 0
+    for bi in cpa.array
+        n += dim(bi)
+        ms = blocks_linear_map(cpa, M, i, n)
+        push!(array, overapproximate(ms, oa))
+        i += n
+    end
+
+    result = CartesianProductArray(array)
+    return result
+end
+
+
+function blocks_linear_map(cpa::CartesianProductArray{N,<:LazySet{N}},
+                                    M::AbstractMatrix{N}, row_start_ind::Int, row_end_ind::Int) where {N}
+    col_start_ind, col_end_ind = 1, 0
+    h_min_sum = MinkowskiSumArray(length(cpa.array), N)
```

Comment:
I would prefer to allocate a `Vector{LazySet{N}}` with a `sizehint!` and in the end return a `MinkowskiSumArray(h_min_sum)`, just as you do it for the `CPA` above.